packages/api-explorer/src/form-components/SchemaField.jsx

```diff
@@ -70,13 +71,36 @@ function getTypeLabel(schema) {
 }
 
 function CustomTemplate(props) {
-  const { id, classNames, label, help, required, description, errors, children, schema } = props;
+  const {
+    id,
+    classNames,
+    label,
+    help,
+    required,
+    description,
+    errors,
+    children,
+    schema,
+    onKeyChange,
+  } = props;
+
+  let editableLabel = label;
+  if (ADDITIONAL_PROPERTY_FLAG in schema) {
+    editableLabel = (
+      <input
+        defaultValue={label}
+        id={`${id}-key`}
+        onBlur={event => onKeyChange(event.target.value)}
+        type="text"
+      />
+    );
+  }
```

Comment:
Can we refactor this to move the conditional in to the JSX template?

```jsx
<label className="label-name" htmlFor={id}>
  {ADDITIONAL_PROPERTY_FLAG in schema ? EditLabel : label}
  // etc.
</label>
```

That way we can simplify the preceding `editableLabel` snippet:

```js
const EditLabel = <input
  defaultValue={label}
  id={`${id}-key`}
  onBlur={event => onKeyChange(event.target.value)}
  type="text"
/>;
```

Reply:
(obviously not a functional concern; don't hold your merge over it or anything!)